src/main/webapp/resources/scripts/controllers/RootController.coffee

```diff
@@ -8,7 +8,13 @@ angular.module('calendar').controller 'RootController', ($rootScope, $scope, Eve
       event:
         color: '#000000'
 
+  eventCache = null
   $scope.collapseForm = ->
+    index = $rootScope.events.indexOf($scope.form.event)
+    if index != -1
+      $rootScope.events[index] = angular.copy(eventCache)
+    eventCache = null
```

Comment:
Now even if you save event it is restored from cache (you have to refresh to see any changes). Maybe it will be easier to store a copy of an event in a form and apply it on save?